From f1caf69a41659a4bc1de643ee2528867b5a47535 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Tue, 10 Sep 2013 22:24:46 -0500
Subject: [PATCH 182/390] rtl8192du: Fix possible buffer overruns in
 core/rtw_mlme_ext.c

These problems were reported by smatch.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_mlme_ext.c    | 1 -
 include/rtw_mlme_ext.h | 1 +
 include/sta_info.h     | 2 +-
 3 files changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
+++ b/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
@@ -1810,7 +1810,6 @@ unsigned int OnAssocReq(struct rtw_adapt
 			if (pstapriv->sta_aid[pstat->aid - 1] == NULL)
 				break;
 
-		/* if (pstat->aid > NUM_STA) { */
 		if (pstat->aid > pstapriv->max_num_sta) {
 			pstat->aid = 0;
 
--- a/drivers/staging/rtl8192du/include/rtw_mlme_ext.h
+++ b/drivers/staging/rtl8192du/include/rtw_mlme_ext.h
@@ -895,6 +895,7 @@ static struct fwevent wlanevents[] =
 	{0, NULL},
 	{0, NULL},
 	{0, &rtw_cpwm_event_callback},
+	{0, NULL},
 };
 
 #endif/* _RTL8192C_CMD_C_ */
--- a/drivers/staging/rtl8192du/include/sta_info.h
+++ b/drivers/staging/rtl8192du/include/sta_info.h
@@ -370,7 +370,7 @@ struct	sta_priv {
 	 * AID is in the range 1-2007, so sta_aid[0] corresponders to AID 1
 	 * and so on
 	 */
-	struct sta_info *sta_aid[NUM_STA];
+	struct sta_info *sta_aid[NUM_STA + 1];
 
 	u16 sta_dz_bitmap;/* only support 15 stations, staion aid bitmap for sleeping sta. */
 	u16 tim_bitmap;/* only support 15 stations, aid=0~15 mapping bit0~bit15 */
